proto/types/canonical.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+package tendermint.proto.types;
+
+option go_package = "github.com/tendermint/tendermint/proto/types";
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "proto/types/types.proto";
+import "google/protobuf/timestamp.proto";
+
+message CanonicalBlockID {
+  bytes                  hash         = 1;
+  CanonicalPartSetHeader parts_header = 2 [(gogoproto.nullable) = false];
+}
+
+message CanonicalPartSetHeader {
+  bytes  hash  = 1;
+  uint32 total = 2;
+}
+
+message CanonicalProposal {
+  SignedMsgType             type      = 1;  // type alias for byte
+  int64                     height    = 2 [(gogoproto.moretags) = "binary:\"fixed64\""];
+  int64                     round     = 3 [(gogoproto.moretags) = "binary:\"fixed64\""];
+  int64                     pol_round = 4 [(gogoproto.moretags) = "binary:\"fixed64\"", (gogoproto.customname) = "POLRound"];
```

Comment:
What's with the fixed64?

Reply:
these were used for amino because the time was set as int and then it needed to be forced to fixed64 for amino encoding. It has been removed